onegov/election_day/formats/election/wabstic_proporz.py

```diff
@@ -69,34 +69,99 @@ def line_is_relevant(line, number, district=None):
 
 
-def get_entity_id(line, entities):
-    entity_id = int(line.bfsnrgemeinde or 0)
-    return 0 if entity_id in EXPATS else entity_id
-
-
-def get_candidate_id(line):
-    return str(int(line.knr))
+def get_number_of_mandates(line):
+    col = 'sitze'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.sitze or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_ausmittlungsstand(line):
+    col = 'ausmittlungsstand'
+    try:
+        t = int(line.ausmittlungsstand or 0)
+        assert 0 <= t <= 3
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+    except AssertionError:
+        raise AssertionError(
+            _('Value ${col} is not between 0 and 3', mapping={'col': col}))
+
+
+def get_stimmberechtigte(line):
+    col = 'stimmberechtigte'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.stimmberechtigte or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_stimmentotal(line):
+    col = 'stimmentotal'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.stimmentotal)
+    except ValueError:
+        raise ValueError(_('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_entity_id(line, expats):
+    if not hasattr(line, 'bfsnrgemeinde'):
+        raise ValueError(
+            _('Missing column: bfsnrgemeinde'))
+    try:
+        entity_id = int(line.bfsnrgemeinde or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: bfsnrgemeinde'
+              ' Can not extract entity_id.'))
+    return 0 if entity_id in expats else entity_id
+
+
+def get_list_id_from_knr(line):
+    """
+    Takes a line from csv file with a candidate number (knr) in it and
+    returns the derived listnr for this candidate. Will also handle the new
+    WabstiC Standard 2018.
+    """
+    col = 'knr'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col', col}))
+    if '.' in line.knr:
+        return line.knr.split('.')[0]
+    # replaces int(int(line.knr) / 100))
+    return line.knr[0:-2]
 
 
 def get_list_id(line):
-    if hasattr(line, 'listnr'):
-        number = int(line.listnr or 0)
-    else:
-        number = int(int(get_candidate_id(line)) / 100)
-    number = 999 if number == 99 else number  # blank list
-    return str(number)
+    col = 'listnr'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    number = line.listnr or '0'
+    number = '999' if number == '99' else number  # blank list
```

Comment:
I'm not quite sure what `# blank list` means. Could you elaborate your comment here? Why is 99 turned into 999?

Reply:
Blank lists are lists without a candidate or those with a missing party. WabstiC uses code 99 for those, but they map to 999 that represents the same from other interfaces. This can read in `open_data_de.md`. I made the comment more clear referencing `open_data_de.md`.